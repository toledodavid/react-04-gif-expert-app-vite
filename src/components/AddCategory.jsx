import { useState } from 'react';


export const AddCategory = () => {

  const [inputValue, setInputValue] = useState('Naruto');

  const onInputChange = ({target}) => {
    setInputValue(target.value);
  }

  const onSubmit = (event) => {
    event.preventDefault();
    console.log(inputValue);
  }

  return(
    <form onSubmit={event => onSubmit(event)}>
      <input type="text"
        placeholder="Search gifs"
        value={inputValue}
        onChange={onInputChange} />
    </form>
  );
}