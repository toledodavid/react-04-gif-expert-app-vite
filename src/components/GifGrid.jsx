import { useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';


export const GifGrid = ({category}) => {

  useEffect(() => {
    getGifs();
  }, []);

  return(
    <>
      <h3>{category}</h3>
      <p>Hello world</p>
    </>
  );
}