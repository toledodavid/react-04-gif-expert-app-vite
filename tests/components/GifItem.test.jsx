import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Tests for <GifItem />', () => {

  const title = 'Saitama';
  const url = 'https://one-punch.com/img.jpg';

  test('should match with the snapshot', () => {
    const {container} = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

});