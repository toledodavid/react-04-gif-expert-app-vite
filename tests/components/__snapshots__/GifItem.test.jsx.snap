// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/img.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
